core/camel-support/src/main/java/org/apache/camel/support/MessageSupport.java

```diff
@@ -204,8 +204,6 @@ public void copyFromWithNewBody(Message that, Object newBody) {
             return;
         }
 
-        // must copy over CamelContext
-        CamelContextAware.trySetCamelContext(that, camelContext);
```

Comment:
The code is already setting up the context in the caller methods (either in the `copy` or in the caller methods prior to this one).